### PR TITLE
Add some docs

### PR DESCRIPTION
**Pull request by derekdreery**
Also make some changes that should have no effect on functionality, but aid readability.

### CODE REVIEW

File: src/cell.rs [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 use std::cell::UnsafeCell;
 use std::error::Error;
 use std::fmt::{Display, Error as FormatError, Formatter};
 use std::ops::{Deref, DerefMut};
 use std::sync::atomic::{AtomicUsize, Ordering};
+use std::usize;
 
+/// Marker struct for an invalid borrow error
 #[derive(Clone, Copy, Debug)]
-pub struct InvalidBorrow;
```
Comment (derekdreery) [inline src/cell.rs]: This seems to be an implementation detail, since problems always result in panics.

File: src/cell.rs [hunk 2]
```diff
@@ -24,6 +24,9 @@ impl Error for InvalidBorrow {
     }
 }
 
+/// An immutable reference to data in a `TrustedCell`.
+///
+/// Access the value via. `std::ops::Deref` (e.g. `*val`)
```
Comment (torkleyy) [inline src/cell.rs]: Please remove the full stop here.

File: src/cell.rs [hunk 3]
```diff
@@ -44,6 +47,9 @@ impl<'a, T> Drop for Ref<'a, T> {
     }
 }
 
+/// A mutable reference to data in a `TrustedCell`.
+///
+/// Access the value via. `std::ops::DerefMut` (e.g. `*val`)
```
Comment (torkleyy) [inline src/cell.rs]: Same here

File: src/cell.rs [hunk 4]
```diff
@@ -24,6 +24,9 @@ impl Error for InvalidBorrow {
     }
 }
 
+/// An immutable reference to data in a `TrustedCell`.
```
Comment (torkleyy) [inline src/cell.rs]: `TrustCell`

File: src/cell.rs [hunk 5]
```diff
@@ -44,6 +47,9 @@ impl<'a, T> Drop for Ref<'a, T> {
     }
 }
 
+/// A mutable reference to data in a `TrustedCell`.
```
Comment (torkleyy) [inline src/cell.rs]: Same here

File: src/cell.rs (hunk 1 above)
Comment (torkleyy) [reply src/cell.rs]: :thinking: Maybe we should add `try_*` variants?

Comment (torkleyy) [reply src/cell.rs]: As you mentioned, that would be helpful for FFI

Comment (derekdreery) [reply src/cell.rs]: I'll add the try variants :)

**Comment by derekdreery** [comment]
I've made the changes you suggested. Because I needed (Partial)Eq implementations for some tests, I just implemented all standard traits. I think they are the ones that are recommended from the libz blitz :)

File: src/cell.rs [hunk 6]
```diff
@@ -120,9 +207,30 @@ impl<T> TrustCell<T> {
...
+        Ok(RefMut {
+            flag: &self.flag,
+            value: unsafe { &mut *self.inner.get() },
+        })
+    }
+
+    /// Gets read-only access to the inner value, bypassing the Cell.
+    ///
+    /// Absence of write access is checked at compile time.
+    pub fn get(&self) -> &T {
```
Comment (torkleyy) [inline src/cell.rs]: That doesn't quite work. There could be a mutable borrow (using `borrow_mut`).

**Comment by derekdreery** [comment]
I've left all the trait impls on the error type, because it makes composing errors easier, if you want to test those for equality.

**Comment by derekdreery** [comment]
:D

**Pull request merged**